lib/client.js

```diff
@@ -258,7 +258,7 @@ Client.prototype.close = function (done) {
           }
         }
       })
-      that.will = null // this function might be called twice
+      that._finished = true // this function might be called twice
```

Comment:
a better solution is to extract `that.will` into a variable, and keep that into the context of the closure.

Reply:
@mcollina Thanks you, sir!